pandas/tests/indexes/common.py

```diff
@@ -795,6 +795,12 @@ def test_inv(self, simple_index):
             with pytest.raises(TypeError, match=msg):
                 ~Series(idx)
 
+    def test_is_categorical_is_deprecated(self, simple_index):
+        # GH50042
+        idx = simple_index
+        with tm.assert_produces_warning(FutureWarning):
```

Comment:
can we check the warning message too please? use `match=` (check ksome other tests)

Reply:
+1

Reply:
Could you explain this a bit more?

Reply:
Yes.

Reply:
This ensures we capture the correct error.